service.subtitles.traduttorianonimi/resources/lib/TraduttoriAnonimi.py

```diff
@@ -0,0 +1,157 @@
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+#  MA 02110-1301, USA.
+#  
+# 
+import xbmc;
+class LogStream(object):
+    def write(self,data):
+        xbmc.log(("""*** [TraduttoriAnonimiCore] -> {}""".format(data)).encode('utf-8'), level = xbmc.LOGNOTICE)
+            
+from bs4 import BeautifulSoup;
+import requests;
+import re;
+import logging;
+import urlparse;
+import traceback
+        
+log=logging.getLogger("TraduttoriAnonimiCore");
+log.setLevel(logging.DEBUG);
+style=logging.Formatter("%(asctime)s {%(levelname)s} %(name)s.%(funcName)s() -->> %(message)s");
+consoleHandler = logging.StreamHandler(LogStream());
+consoleHandler.setFormatter(style)
+log.addHandler(consoleHandler);
+
+def MagicInt(x): #Cast to integer (if possible) without Errors
+    try:
+        return int(x);
+    except:
+        return x;
+
+def nstring(s):
+    try:
+        return s.encode(errors="replace").replace("?"," ")
+    except:
+        try:
+            return s.decode(errors="replace").replace(u"\ufffd\ufffd\ufffd"," ")
+        except:
+            return s;
+
+class SearchableDict(dict):
+    def __getitem__(self, token):
+        p=re.compile(token,re.IGNORECASE)
+        result=[];
+        for x,y in self.items():
+            if re.search(p,nstring(x)):
+                result.append({"Name":x,"URL":y});
+                
+        if (len(result)==0):
+            result=None;
+        
+        return result;
+        
+class SearchableList(list):
+    def __getitem__(self, token):
+        p=re.compile(token,re.IGNORECASE)
+        result=[];
+        for x in self:
+            if re.search(p,x):
+                result.append(x);
+                
+        if (len(result)==0):
+            result=None;
+        
+        return result;
+        
+
+def RetriveURL(url):
+    try:
+        headers={'user-agent': 'Kodi-SubtitleService-TraduttoriAnonimi'};
+        q=requests.get(url,headers=headers);
+        return q;
+    except Exception as exc:
+        log.error(traceback.format_exc())
+        raise exc;
+
+
+
+class TraduttoriAnonimi:
+    def __init__(self,BaseURL="http://traduttorianonimi.it"):
+        self.log=log.getChild("TA")
+        
+        self.BaseURL=BaseURL;
+        self.log.debug("BaseURL=> {}".format(self.BaseURL))
+        
+        self.ShowsListURL=urlparse.urljoin(self.BaseURL,"elenco-serie/")
+        self.log.debug("ShowsListURL=> {}".format(self.ShowsListURL))
+        
+        self.ShowsList=SearchableDict()
+        self.UpdateShowsList();
+        
+    def UpdateShowsList(self):
+        self._GrabShowsListFromWebsite();
+        
+    def _GrabShowsListFromWebsite(self):
+        self.log.info("Grabbing Shows list from Website.")
+        url=self.ShowsListURL
+        self.log.debug("self.ShowsList re-initialized")
+        self.ShowsList=SearchableDict();
+        while (1):
```

Comment:
i'm fixing right now. with a (MAX_RETRIES=100 while (MAX_RETRIES>=0)) have you any advice before i commit?